www/js/push.js

```diff
@@ -60,14 +61,46 @@ function createPushDevice(registrationId) {
 }
 
 // Delete a push device belonging to this user
-function deletePushDevice(registrationId) {
+function deletePushDevice(registrationId, callback) {
   $.ajax({
     url: API + '/push_devices',
     type: 'DELETE',
     dataType: 'json',
     data: {token: registrationId},
     error: function(resp) {
       alert("Failed to delete push device id!");
+    },
+    always: function(resp) {
+      if(callback) callback();
     }
   });
 }
+
+// Send push device to server after sign in
+function pushAfterLogin() {
+  var registrationId = localStorage.getItem('registrationId');
+  if(registrationId !== null) updatePushDevice(null, registrationId);
+}
+
+function deletePushAndSignOut() {
+  var registrationId = localStorage.getItem('registrationId');
+
+  if (registrationId !== null) {
+    // Remove push device from the server
+    deletePushDevice(registrationId, $.auth.signOut);
+
+    // Unregister from firebase
+    pushService.unregister(function() {
+      // Success, all event listeners removed. Init the push service again
+      setupPush();
+    }, function() {
+      // Error
+      alert('Failed to unregisted from push service!');
+    });
+  } else {
+    $.auth.signOut();
+  }
+}
```

Comment:
Yes, we sign out in either case. But the ajax request to remove the push device must finish before we sign out, otherwise we will get "401 Unauthorized", and cannot remove the PD.